gef-ui/server/controllers/amendments/do-you-have-a-facility-end-date/do-you-have-a-facility-end-date.ts

```diff
@@ -0,0 +1,39 @@
+import { CustomExpressRequest } from '@ukef/dtfs2-common';
+import { Response } from 'express';
+import * as api from '../../../services/api';
+import { DoYouHaveAFacilityEndDateViewModel } from '../../../types/view-models/amendments/do-you-have-a-facility-end-date-view-model';
+import { asLoggedInUserSession } from '../../../utils/express-session';
+import { userCanAmendFacility } from '../../../utils/facility-amendments.helper';
+
+export type GetDoYouHaveAFacilityEndDateRequest = CustomExpressRequest<{
+  params: { dealId: string; facilityId: string; amendmentId: string };
+}>;
+
+export const getDoYouHaveAFacilityEndDate = async (req: GetDoYouHaveAFacilityEndDateRequest, res: Response) => {
+  try {
+    const { dealId, facilityId, amendmentId } = req.params;
+    const { userToken, user } = asLoggedInUserSession(req.session);
+
+    const deal = await api.getApplication({ dealId, userToken });
+    const { details: facility } = await api.getFacility({ facilityId, userToken });
+
+    if (!deal || !facility) {
+      return res.redirect('/not-found');
+    }
+
+    if (!userCanAmendFacility(facility, deal, user.roles)) {
+      return res.redirect(`/gef/application-details/${dealId}`);
```

Comment:
Console.error please.

Reply:
Whoever picks this up, we should update other amendment controllers to do the same, to be consistent.